### ISSUE TITLE
Separate Metadata address

### ISSUE THREAD

**Issue by rpmansion** [Sustainsys/Saml2#1185]
Currently, we do not have a separate address path to view the metadata. Thus I have to define the `ModulePath` something like the below to do this which I know is being used as Saml2 base endpoints.

```
var spOptions = new SPOptions
{
       // . . .
	ModulePath = "saml2/sp/metadata",
        EntityId = new EntityId($"{baseUrl}/saml2/sp"),
};
```

The reason why I did this is some federations have the policy for naming the entity id like the [UK Federation](https://ukfederation.org.uk/content/Documents/EntityIDPolicy). However checking the metadata structure you would see the below parts.

```
<SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://vvimpacttracker.com/saml2/sp/metadata/Logout" />
<AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://vvimpacttracker.com/saml2/sp/metadata/Acs" index="0" isDefault="true" />
<AssertionConsumerService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact" Location="vvimpacttracker.com/saml2/sp/metadata/Acs" index="1" isDefault="false" />
```

So if we could have another option to set the address for the metadata and instead of the browser downloading the file is to view this in the browser as XML?

**Comment by AndersAbel** [comment]
Yes, better endpoint naming/routing would be good, see e.g. #1031 for a request to rename Acs.

What I think is important is to offer a configuration that by default makes people have the same Uri for EntityId and the metadata adress. It is recommended by the spec, but I know that there are many cases where that recommendation is not follwoed.
